giskard/ml_worker/exceptions/giskard_exception.py

```diff
@@ -1,3 +1,40 @@
 class GiskardException(Exception):
     def __init__(self, *args: object) -> None:
         super().__init__(*args)
+
+
+class GiskardPythonEnvException(GiskardException):
+    def __init__(self, *args: object) -> None:
+        super().__init__(*args)
+
+
+class GiskardPythonVerException(GiskardPythonEnvException):
+    def __init__(self, cls_name, e, *args: object) -> None:
+        super().__init__(
+            f"Failed to load '{cls_name}' due to {e.__class__.__name__}.\n"
+            "Make sure you are loading it in the environment with matched Python version.",
+            *args,
+        )
+
+
+class GiskardPythonDepException(GiskardPythonEnvException):
+    def __init__(self, cls_name, e, *args: object) -> None:
+        super().__init__(
+            f"Failed to load '{cls_name}' due to {e.__class__.__name__}.\n"
+            "Make sure you are loading it in the environment with matched dependencies.",
+            *args,
+        )
+
+
+def python_env_exception_helper(cls_name, e: Exception):
+    if isinstance(e, TypeError):
+        # Python 3.9: code() takes at most 16 arguments (18 given)
+        # Python 3.10: code expected at most 16 arguments, got 18
+        # Python 3.11: code() argument 13 must be str, not int
+        if "at most 16 arguments" in e.args[0] or "code() argument 13 must be str, not int" in e.args[0]:
```

Comment:
We should probably compare the Python version meta with the current python version to be independent of the format (ei. Python 3.12 might break this code)

Reply:
With the version meta, it should be feasible

Reply:
Do you think whether we should make another PR for the Python versions of the callable function?
I am drafting one for the Giskard Hub DB and backend changes: https://github.com/Giskard-AI/giskard-hub/pull/256

Reply:
Not sure if it's worth since we will probably get ride of the `pickle` serialization to just saving python code.

Reply:
OK, let me keep this way at the moment. And add the Python version to the code later